modules/local/corroborate_umis.nf

```diff
@@ -0,0 +1,76 @@
+process CORROBORATE_UMIS {
+    tag "$meta.id"
+    label "process_low"
+
+    conda (params.enable_conda ? "conda-forge::python=3.9.5" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
+        'quay.io/biocontainers/python:3.9--1' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.fastq.gz"),        emit: reads
+    tuple val(meta), val(prefix), val(umiok),   emit: umistatus
+    path "versions.yml"                     ,   emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    filename = reads.toString()
+    if (meta.single_end) {
+        """
+
+        prefix=${prefix}
+        # [[ \$(file --mime-type -b "$reads") == "text/plain" ]] would be better, but file is not part of the container.
```

Comment:
This is getting quite difficult to follow and could quite easily break in the long run. I am wondering whether it might be a good idea to add a small Python script to do this instead? How would you feel about just adding the main `--skip_umi_extract` functionality in this PR which you have pretty much done already, and then adding this auto-detection in a follow up PR?

Reply:
Commenting now to save you some time with all of the bash syntax testing :)

Reply:
I am not sure if using Python will simplify anything here, but I have indeed found many notations that run fine in a pure bash script, not when compiled into a module ;-)  

I agree that this module is not strictly necessary for now. I will remove it and squash the respective commits.